src/containers/Tenant/Diagnostics/TenantOverview/TenantCpu/TopNodesByCpu.tsx

```diff
@@ -25,27 +15,18 @@ interface TopNodesByCpuProps {
 }
 
 export function TopNodesByCpu({path, additionalNodesProps}: TopNodesByCpuProps) {
-    const dispatch = useTypedDispatch();
-
     const query = useSearchQuery();
 
-    const {wasLoaded, loading, error} = useTypedSelector((state) => state.topNodesByCpu);
     const {autorefresh} = useTypedSelector((state) => state.schema);
-    const topNodes = useTypedSelector(selectTopNodesByCpu);
     const columns = getTopNodesByCpuColumns(additionalNodesProps?.getNodeRef);
 
-    const fetchNodes = React.useCallback(
-        (isBackground: boolean) => {
-            if (!isBackground) {
-                dispatch(setDataWasNotLoaded());
-            }
-
-            dispatch(getTopNodesByCpu({tenant: path}));
-        },
-        [dispatch, path],
+    const {currentData, isFetching, error} = topNodesApi.useGetTopNodesQuery(
+        {tenant: path, sortValue: 'CPU'},
+        {pollingInterval: autorefresh ? DEFAULT_POLLING_INTERVAL : 0},
     );
 
-    useAutofetcher(fetchNodes, [fetchNodes], autorefresh);
+    const loading = isFetching && currentData === undefined;
```

Comment:
Here and further: isn't it possible to use `isLoading` property from query?

Reply:
it repeats the current behaviour. `isLoading` is false when the query arguments change.